IPython/core/magics/namespace.py

```diff
@@ -200,6 +203,10 @@ def psearch(self, parameter_s=''):
         Show objects beginning with a single _::
 
           %psearch -a _*         list objects beginning with a single underscore
+          
+        List available objects::
+        
+          %psearch -l            list all available object types
         """
```

Comment:
Maybe dumb question, as you are the one having the use case; would it make more sens to actually add the list of `typestr2type` to the docstring ?

Reply:
We could just put there, but then again it may change again in the future. I thought more practical to list the *built-ins*.